scriptworker/test/__init__.py

```diff
@@ -269,27 +269,45 @@ def tmpdir2():
 
 
 @pytest.mark.asyncio
-@pytest.yield_fixture(scope='function', params=['firefox'])
-async def rw_context(request, event_loop):
+@pytest.yield_fixture(scope='function')
+async def rw_context(event_loop):
+    with tempfile.TemporaryDirectory() as tmp:
+        context = _craft_rw_context(tmp, event_loop, cot_product='firefox')
+        yield context
+        await _close_context(context)
+
+
+@pytest.mark.asyncio
+@pytest.yield_fixture(scope='function')
+async def mobile_rw_context(event_loop):
     with tempfile.TemporaryDirectory() as tmp:
-        config = get_unfrozen_copy(DEFAULT_CONFIG)
-        config['cot_product'] = request.param
-        context = Context()
-        context.config = apply_product_config(config)
-        context.config['gpg_lockfile'] = os.path.join(tmp, 'gpg_lockfile')
-        context.config['cot_job_type'] = "signing"
-        for key, value in context.config.items():
-            if key.endswith("_dir"):
-                context.config[key] = os.path.join(tmp, key)
-                makedirs(context.config[key])
-            if key.endswith("key_path") or key in ("gpg_home", ):
-                context.config[key] = os.path.join(tmp, key)
-        context.config['verbose'] = VERBOSE
-        context.event_loop = event_loop
+        context = _craft_rw_context(tmp, event_loop, cot_product='mobile')
         yield context
-        await _close_session(context)
-        await _close_session(context.queue)
-        await _close_session(context.temp_queue)
+        await _close_context(context)
+
+
+def _craft_rw_context(tmp, event_loop, cot_product):
```

Comment:
Smart. I was a little concerned that changing the Context creation coroutine to a synchronous function might give us more `aiohttp.ClientSession` warnings. If we did start getting these warnings, the following may help address:

* changing `_craft_rw_context` to a coroutine
* defining `context.event_loop` before creating `context.config`, since we [use the `event_loop` to create the session](https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/context.py#L118).

However, I don't see any of these warnings in Travis, so we may be good. Still, if we went back to pytest-xdist or other concurrent tests, defining `event_loop` before the config may still be a good fix to make.

Reply:
Cool! Thanks for the context. I'll keep an eye out for the warnings. In the meantime, I'm landing this patch as is.